CHANGELOG.md

```diff
@@ -4,6 +4,117 @@ All notable changes to Router will be documented in this file.
+
+Nullability rules may remove parts of the response without indicating why. Error messages now indicate which part of the response triggered nullability rules.
+showing which part of the response triggered nullability rules.
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/1537
+
+### router now provides TraceId ([PR #1663](https://github.com/apollographql/router/issues/1536))
+
+If you need a reliable way to link together the various stages of pipeline processing, you can now use
+
+```rust
+apollo_router::tracer::TraceId
+```
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/1663
+
+## 🐛 Fixes
+
+### Docker images: Use absolute path for `ENTRYPOINT` ([PR #1684](https://github.com/apollographql/router/pull/1684))
+
+This restores the absolute path in `ENTRYPOINT` in our `Dockerfile`s (and published images) to allow users to change their working directory without consequence (and without needing to change it back to `/dist` or override the `entrypoint`).
+
+By [@110y](https://github.com/110y) in https://github.com/apollographql/router/pull/1684
+
+### Update our helm documentation to illustrate how to use our registry ([#1643](https://github.com/apollographql/router/issues/1643))
+
+The helm chart never used to have a registry, so our docs were really just placeholders. I've updated them to reflect the fact that we now store the chart in our OCI registry.
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/1649
+
+### Update router-bridge to `query-planner` v2.1.1 ([PR #1650](https://github.com/apollographql/router/pull/1650) [PR #1672](https://github.com/apollographql/router/pull/1672))
+
+The 2.1.0 release of the query planner comes with fixes to fragment interpretation and reduced memory usage.
+The 2.1.1 release of the query planner fixes an issue with the `@defer` directive's `if` argument being ignored.
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/1650 and https://github.com/apollographql/router/pull/1672
+
+### Do not nullify the entire query if the root operation is not present ([PR #1674](https://github.com/apollographql/router/issues/1674))
+
+If a root field was not returned by the subgraph (e.g., when there's an error) we should not nullify the entire data object. Instead, it's the root field that should be null (unless it is non nullable).
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/1674
+
+### Propagate graphql response regardless of the subgraph HTTP status code. ([#1664](https://github.com/apollographql/router/issues/1664))
+
+Subgraph service calls used to return an error when the received HTTP status code isn't 200.  There's, however, no mention in the GraphQL specification that leads us to assume any intent behind the HTTP status code returned by a GraphQL server since the GraphQL specification is transport agnostic.
```

Comment:
```suggestion
```suggestion
Subgraph service calls no longer return an error when the received HTTP status code isn't 200.  The GraphQL specification does not specify HTTP status code behavior since the GraphQL specification is transport agnostic.
```
```